tools/jvlink-bridge/bridge_native.c

```
@@ -0,0 +1,500 @@
+            if (code >= 0) g_is_open = 1;
+            json_response("{\"status\":\"%s\",\"code\":%d,\"readcount\":%d}",
+                         code >= 0 ? "ok" : "error", code, code >= 0 ? code : 0);
+            free(dataspec); free(key);
+        }
+        else if (strcmp(cmd, "read") == 0) {
+            if (!g_jvlink || !g_is_open) { json_response("{\"status\":\"error\",\"error\":\"not open\"}"); free(cmd); continue; }
+            int size = json_get_int(line, "size", 110000);
+            char* buff = (char*)calloc(size + 1, 1);
+            char filename[512] = "";
+            int code = CallJVRead(buff, size, filename, sizeof(filename));
+            if (code > 0 && buff[0]) {
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)